.github/workflows/tests-runner.yml

```diff
@@ -0,0 +1,195 @@
+# Executes all the tests on all the platforms
+name: Tests
+
+on:
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  # Duplicated jobs so their dependencies are not blocked on both the
+  # setup jobs
+
+  generate_tests_matrix:
+    name: Generate test runsheet
+    runs-on: windows-latest
```

Comment:
In my experience these setup steps are better done on ubuntu just to run faster.

Reply:
There were some weird failures running this on Ubuntu. The only observable difference in my experience is the time to provision a runner on AzDO. I haven't observed any significant difference in GHA.